core/schema/modulesource.go

```diff
@@ -717,16 +736,28 @@ func (s *moduleSourceSchema) loadBlueprintModule(
 		return fmt.Errorf("failed to get dag server: %w", err)
 	}
 
-	// If we have a blueprint module, load it
-	pcfg := src.ConfigBlueprint
-	if pcfg == nil {
-		return nil
+	// Load blueprint
+	if src.ConfigBlueprint != nil {
+		blueprint, err := core.ResolveDepToSource(ctx, bk, dag, src, src.ConfigBlueprint.Source, src.ConfigBlueprint.Pin, src.ConfigBlueprint.Name)
+		if err != nil {
+			return fmt.Errorf("failed to resolve blueprint to source: %w", err)
+		}
+
+		src.Blueprint = blueprint
 	}
-	blueprint, err := core.ResolveDepToSource(ctx, bk, dag, src, pcfg.Source, pcfg.Pin, pcfg.Name)
-	if err != nil {
-		return fmt.Errorf("failed to resolve dep to source: %w", err)
+
+	// Load toolchains array
+	if len(src.ConfigToolchains) > 0 {
+		src.Toolchains = make([]dagql.ObjectResult[*core.ModuleSource], len(src.ConfigToolchains))
+		for i, pcfg := range src.ConfigToolchains {
+			toolchain, err := core.ResolveDepToSource(ctx, bk, dag, src, pcfg.Source, pcfg.Pin, pcfg.Name)
+			if err != nil {
+				return fmt.Errorf("failed to resolve toolchain to source: %w", err)
+			}
+			src.Toolchains[i] = toolchain
+		}
```

Comment:
nit: if straightforward this seems like something worth parallelizing (probably with `util/parallel`, though I think you'll want [the same fix](https://github.com/dagger/dagger/pull/11211/commits/068e00b0338974091b591dfa5b367f72e4b62fea) that's in the checks PR)